### PR TITLE
UserAuthentication-Devise PR #5

### PR DESCRIPTION
**Pull request by NihilMaster**
This is a Pull Request I created to review the changes I made when applying user authentication with Devise.

- [Devise Guide](http://guides.railsgirls.com/devise)
- Adding `devise_for :users` in routes file.
- [Create modifiable Views](https://dev.to/n350071/customize-view-of-devise-3k95)
- Fix `Cancel my account` button.

### CODE REVIEW

File: app/views/users/registrations/edit.html.erb [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      <h3>Cancel my account</h3>
+      <p class="text-warning">Unhappy?</p>
+      <div style="width:100%; display: flex;">
+        <%= link_to "Cancel my account", registration_path(resource_name), data: { confirm: "Are you sure?" }, method: :delete, :class => 'btn btn-outline-danger', :style =>'margin:5px' %>
+        <%= link_to "Back", :back, :class => 'btn btn-outline-secondary', :style =>'margin:5px' %>
+      </div>
+    </div>
+
+  </div>
+</div>
```
Comment (mauriciosabogal) [inline app/views/users/registrations/edit.html.erb]: Please remember the coding styles. Always leave an empty line at the end of the file.
(This applies for the other files too)

File: app/views/users/sessions/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,34 @@
+<div style="width:100%; display: flex; justify-content: center;">
+  <div class="card border-primary" style="width:50%; padding:10px; ">
+    <h2>Log in</h2><!-- Iniciar sesión -->
+
+    <%= form_for(resource, as: resource_name, url: session_path(resource_name)) do |f| %>
```
Comment (mauriciosabogal) [inline app/views/users/sessions/new.html.erb:5]: What is the purpose of this line?

File: config/routes.rb [hunk 3]
```diff
@@ -1,4 +1,5 @@
 Rails.application.routes.draw do    # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+  devise_for :users
```
Comment (mauriciosabogal) [inline config/routes.rb:2]: What does `devise_for :users` mean?

File: yarn.lock [hunk 4]
```diff
@@ -885,6 +885,11 @@
   resolved "https://registry.yarnpkg.com/@csstools/convert-colors/-/convert-colors-1.4.0.tgz#ad495dc41b12e75d588c6db8b9834f08fa131eb7"
   integrity sha512-5a6wqoJV/xEdbRNKVo6I4hO3VjyDq//8q2f9I6PBAvMesJHFauXDorcNCsr9RzvsZnaWi5NYCcfyqP1QeFHFbw==
 
+"@discoveryjs/json-ext@^0.5.0":
```
Comment (mauriciosabogal) [inline yarn.lock]: It looks like you are using `npm` and `yarn` at the same time. That's not a good practice. Please use only one of them (I recommend using `yarn`).

File: app/views/users/registrations/edit.html.erb (hunk 1 above)
Comment (NihilMaster) [reply app/views/users/registrations/edit.html.erb]: **Fixed**. I trusted that my editor was fulfilling that function.

File: config/routes.rb (hunk 3 above)
Comment (NihilMaster) [reply config/routes.rb:2]: It is in charge of creating all the necessary routes for the User model.

File: yarn.lock (hunk 4 above)
Comment (NihilMaster) [reply yarn.lock]: **Fixed**. I chose to use Yarn.

File: app/views/users/sessions/new.html.erb (hunk 2 above)
Comment (NihilMaster) [reply app/views/users/sessions/new.html.erb:5]: It is quite confusing. I know that it is a deprecated method that acts as form_with, which in this line receives as a parameter a Model and that in turn gathers the values to validate a login. I know the session_path creates a URL helper, but I don’t quite understand what resource is. I know it gives the name resourse_name. 
I looked a lot at it without finding anything concise I would like you to explain it to me if possible.

File: app/models/comment.rb [hunk 5]
```diff
@@ -1,4 +1,5 @@
 class Comment < ApplicationRecord
   include Visible
+  # belongs_to :user
```
Comment (mauriciosabogal) [inline app/models/comment.rb]: When you, please make sure there is no commented code.

File: config/routes.rb [hunk 6]
```diff
@@ -1,7 +1,12 @@
...
 
-    resources :articles do
-      resources :comments
-    end
+  resources :articles do
+    resources :comments
+  end
+
+  resources :users do
+    resources :articles
```
Comment (mauriciosabogal) [inline config/routes.rb:10]: You are defining `resources :articles` twice. Please refactor.

File: package.json [hunk 7]
```diff
@@ -1,19 +0,0 @@
-{
```
Comment (mauriciosabogal) [inline package.json:1]: Why did you delete this file?

File: app/models/comment.rb (hunk 5 above)
Comment (NihilMaster) [reply app/models/comment.rb]: **Fixed**

File: package.json (hunk 7 above)
Comment (NihilMaster) [reply package.json:1]: I restored the file and updated the yarn.

**Review by mauriciosabogal — APPROVED** [review]
